splitio/sync/split.py

```diff
@@ -318,3 +354,169 @@ def synchronize_splits(self, till=None):  # pylint:disable=unused-argument
 
         for split in to_delete:
             self._split_storage.remove(split)
+
+        return []
+
+    def _synchronize_json(self):
+        """
+        Update splits in storage for json mode.
+
+        :return: segment names string array
+        :rtype: [str]
+        """
+        try:
+            fetched, till = self._read_splits_from_json_file(self._filename)
+            segment_list = set()
+            fecthed_sha = util._get_sha(json.dumps(fetched))
+            if fecthed_sha != self._current_json_sha:
+                self._current_json_sha = fecthed_sha
+                if self._split_storage.get_change_number() <= till or till == self._DEFAULT_SPLIT_TILL:
+                    for split in fetched:
+                        if split['status'] == splits.Status.ACTIVE.value:
+                            parsed = splits.from_raw(split)
+                            self._split_storage.put(parsed)
+                            _LOGGER.debug("split %s is updated", parsed.name)
+                            segment_list.update(set(parsed.get_segment_names()))
+                        else:
+                            self._split_storage.remove(split['name'])
+
+                    self._split_storage.set_change_number(till)
+            return segment_list
```

Comment:
similar reordering to reduce nestind would be nice here as well

Reply:
will do